File: solutions/javascript/resistor-color-duo/1/resistor-color-duo.js
export const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white",
];

export const decodedValue = (colors) => {
  let value = 0;
  for (const color of colors.slice(0, 2)) {
    value *= 10;
    value += COLORS.indexOf(color);
  }

  return value;
};

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
from enum import Enum
from itertools import islice

class ColorCode(Enum):
    BLACK  = 0
    BROWN  = 1
    RED    = 2
    ORANGE = 3
    YELLOW = 4
    GREEN  = 5
    BLUE   = 6
    VIOLET = 7
    GREY   = 8
    WHITE  = 9

def value(colors: list[str]) -> int:
    first_two = list(islice(colors, 2))
    try:
        return sum(ColorCode[color.upper()].value * (10 ** i) for i, color in enumerate(reversed(first_two)))
    except KeyError as e:
        raise ValueError(f"Unknown color in: {first_two}") from e
